src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs

```diff
@@ -27,6 +27,11 @@ public class AuthorizationOptions
         /// </remarks>
         public AuthorizationPolicy DefaultPolicy { get; set; } = new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build();
 
+        /// <summary>
+        /// Gets or sets the required authorization policy. Defaults to null.
+        /// </summary>
+        public AuthorizationPolicy RequiredPolicy { get; set; }
```

Comment:
So when would I use RequiredPolicy vs DefaultPolicy?

Reply:
DefaultPolicy = run on endpoint with `[Authorize]` (no policy name)
RequiredPolicy = run on ALL endpoints (no authorize)

RequiredPolicy was... required... in the static files sample before I changed it to set an endpoint with IAuthorizeData metadata on it.

Do you have an opinion on whether RequiredPolicy is a useful thing to have? We could remove it.

Reply:
I've seen enough people write their own Authorization middleware that says "Deny all anonymous requests beyond this point", so yes. E.g. It's a global Authorize attribute.

Reply:
Add that explanation to the doc comments please.